pl_bolts/models/detection/components/torchvision_backbones.py

```diff
@@ -0,0 +1,105 @@
+
+from typing import Optional
+
+import torch.nn as nn
+
+from pl_bolts.utils.warnings import warn_missing_pkg
+
+try:
+    import torchvision
+
+    from pl_bolts.models.detection.components._supported_models import TORCHVISION_MODEL_ZOO
+except ModuleNotFoundError:
+    warn_missing_pkg('torchvision')  # pragma: no-cover
+
+
+def _create_backbone_generic(model: nn.Module, out_channels: int):
+    """
+    Generic Backbone creater. It removes the last linear layer.
+
+    Args:
+        model: torch.nn model
+        out_channels: Number of out_channels in last layer.
+    """
+    modules_total = list(model.children())
+    modules = modules_total[:-1]
+    ft_backbone = nn.Sequential(*modules)
+    ft_backbone.out_channels = out_channels
+    return ft_backbone
+
+
+# Use this when you have Adaptive Pooling layer in End.
+# When Model.features is not applicable.
+def _create_backbone_adaptive(model: nn.Module, out_channels: Optional[int] = None):
+    """
+    Creates backbone by removing linear after Adaptive Pooling layer.
+
+    Args:
+        model: torch.nn model with adaptive pooling layer
+        out_channels: Number of out_channels in last layer
+    """
+    if out_channels is None:
+        modules_total = list(model.children())
+        out_channels = modules_total[-1].in_features
+    return _create_backbone_generic(model, out_channels=out_channels)
+
+
+def _create_backbone_features(model: nn.Module, out_channels: int):
+    """
+    Creates backbone from feature sequential block.
+
+    Args:
+        model: torch.nn model with features as sequential block.
+        out_channels: Number of out_channels in last layer.
+    """
+    ft_backbone = model.features
+    ft_backbone.out_channels = out_channels
+    return ft_backbone
+
+
+def create_torchvision_backbone(model_name: str, pretrained: bool = True):
+    """
+    Creates CNN backbone from Torchvision.
```

Comment:
A tiny fix for consistency :]
```suggestion
    Creates CNN backbone from torchvision.
```